app/assets/javascripts/arclight/context_navigation.js

```diff
@@ -55,6 +90,159 @@ class ContextNavigation {
+   * Inserts <li> elements for parents (e. g. components or collections) for the
+   *   current Document and appends <li> for each of these
+   * @param {NavigationDocument[]} newDocs - the NavigationDocument objects for
+   *   each resulting Solr Document
+   * @param {string[]} originalParents - the IDs for the Solr Documents of each
+   *   ancestor
+   * @param {string} parent - the ID for the immediate parent (ancestor)
+   * @param {jQuery} parentLi - the <li> used to generate the <ul> for the
+   * context - this is consistently the *last* element in the <ul>
+   */
+  updateParents(newDocs, originalParents, parent, parentLi) {
+    // Case where this is a parent list and needs to be filed correctly
+    //
+    // Otherwise, retrieve the parent...
+    const parentIndex = originalParents.indexOf(parent);
+
+    // The first parent is always used to consistently construct the doc ID
+    const firstParent = originalParents[0];
+    const nextParent = originalParents[parentIndex + 1];
+    const currentId = `${firstParent}${nextParent}`;
+
+    const newDocIndex = newDocs.findIndex(doc => doc.id === currentId);
+
+    // Update the docs before the item
+    // Retrieves the documents up to and including the "new document"
+    const beforeDocs = newDocs.slice(0, newDocIndex);
+    let renderedBeforeDocs;
+    if (beforeDocs.length > 1) {
+      beforeDocs.forEach(function (parentDoc) {
+        parentDoc.collapse();
+      });
+      renderedBeforeDocs = beforeDocs.map(newDoc => newDoc.render()).join('');
+      const prevParentContainer = this.buildExpandContainer();
+      prevParentContainer.append(renderedBeforeDocs);
+    } else {
+      renderedBeforeDocs = beforeDocs.map(newDoc => newDoc.render()).join('');
+    }
+
+    parentLi.before(renderedBeforeDocs).fadeIn(500);
+
+    let itemDoc = newDocs.slice(newDocIndex, newDocIndex + 1);
+    let renderedItemDoc = itemDoc.map(doc => doc.render()).join('');
+
+    // Update the item
+    const $itemDoc = $(renderedItemDoc);
+    // Update the id, add classes of the classes. Prepend the current children.
+    parentLi.attr('id', $itemDoc.attr('id'));
+    parentLi.addClass($itemDoc.attr('class'));
+
+    parentLi.prepend($itemDoc.children()).fadeIn(500);
+
+    // Update the docs after the item
+    const afterDocs = newDocs.slice(newDocIndex + 1, newDocs.length);
+    const renderedAfterDocs = afterDocs.map(newDoc => newDoc.render()).join('');
+
+    // Insert the documents after the current
+    parentLi.after(renderedAfterDocs).fadeIn(500);
+  }
+
+  /**
+   * Update the ancestors for <li> elements
+   * @param {jQuery} $li - the <li> element for the current, highlighted
+   *   Document in the <ul> context list of collections, components, and
+   *   containers
+   */
+  updateListSiblings($li) {
+    const prevSiblings = $li.prevAll('.al-collection-context');
+    /**
+     * @todo This should be deduplicated and refactored - perhaps a Class
+     *   derived from ExpandButton?
```

Comment:
Refactoring this will include the i18n support, I just wanted to be certain that nothing else in these changes are potentially problematic.